relay-server/src/processing/spans/process.rs

```diff
@@ -35,10 +38,44 @@ pub fn expand(spans: Managed<SampledSpans>) -> Managed<ExpandedSpans> {
 
         spans::integrations::expand_into(&mut all_spans, records, spans.inner.integrations);
 
+        let mut span_id_mapping: BTreeMap<_, _> = all_spans
+            .into_iter()
+            .filter_map(|span| {
+                if let Some(id) = span.value().and_then(|span| span.span_id.value().copied()) {
+                    return Some((id, SpanWrapper::new(span)));
+                }
+                None
+            })
+            .collect();
```

Comment:
So this now made some integration tests fail since it enforces that span_ids are unique. I think this is ok and updated the tests for now, but let me know if we should not be enforcing this.

Reply:
Good catch, we should emit outcomes.

Reply:
Updated this now to drop if we fail to extract the span_id (think that should never happen) or if the span_ids collide (should not happen but could).